### ISSUE TITLE
CEShare fails to log in if one of these characters is used in the FearLess password `=/<}(`

### ISSUE THREAD

**Issue by ScrewTSW** [cheat-engine/cheat-engine#2917]
I've been repeatedly getting `Wrong Password` error until I changed my password to only contain `!@#$%^&*.`

**Comment by fade2gray** [comment]
Maybe you need to escape those characters? e.g. \=\/\\\<\}\(

**Comment by ScrewTSW** [comment]
@fade2gray yeah. I think the actual culprit is missing character escape over here https://github.com/cheat-engine/cheat-engine/blob/54f90a22bfc9d4981f414a7917d70d3d17f190e5/Cheat%20Engine/bin/autorun/ceshare/ceshare_account.lua#L166

the url_encode method seems to work just fine

**Comment by cheat-engine** [comment]
try adding `str=str:gsub("%W", "%%%0")` before [`str = string.gsub (str, "([^%w %-%_%.%~])", `](https://github.com/cheat-engine/cheat-engine/blob/54f90a22bfc9d4981f414a7917d70d3d17f190e5/Cheat%20Engine/bin/autorun/ceshare.lua#L305) and see if that works better

**Comment by ScrewTSW** [comment]
> try adding `str=str:gsub("%W", "%%%0")` before [`str = string.gsub (str, "([^%w %-%_%.%~])", `](https://github.com/cheat-engine/cheat-engine/blob/54f90a22bfc9d4981f414a7917d70d3d17f190e5/Cheat%20Engine/bin/autorun/ceshare.lua#L305) and see if that works better

I can try that (have to figure out how to compile cheatengine from source first), but I think sanitation at input would still be a better call.
In general, you probably want to avoid passing malformed data between modules, and I still have a suspicion something might be happening at the place, where user input is extracted from the form, I'm not sure how Lua handles converting the input field to string.

**Comment by ScrewTSW** [comment]
> you don't need to compile CE for this.  You just edit the lua file and restart ce

Doesn't seem to have helped.
Do you have any personal contact where I could send logs with sensitive data?
I've printed out debug strings throughout the method, but the data contains my username and actual password

**Comment by ScrewTSW** [comment]
URLEncoded parameters also decode properly without an error, the raw POST response is just XML with wrong password, but the URLDecodes password still works for log in through forums page.

I'm not sure what's wrong at this point

**Comment by cheat-engine** [comment]
The fix I posted won't work no.  How it was originally does work for me with the characters I tested with. ( AbCd1234!@#$-=[] )

But try setting your ceshare to https://cheatengine.org/ceshare/ and then try to login.  (you'll first get some other errors as it's not a full implementation)  but it will give feedback that might help

Or post here a similar password you use for testing
